File: 프로그래머스/1/176963. 추억 점수/추억 점수.swift
import Foundation

func solution(_ name:[String], _ yearning:[Int], _ photo:[[String]]) -> [Int] {
    var result: [Int] = []
    var dic: [String: Int] = [:]
    
    for i in name.indices {
        dic.updateValue(yearning[i], forKey: name[i])
    }
    
    photo.forEach { strArr in
        result.append(strArr.map { dic[$0] ?? 0 }.reduce(0, +))
    }
    print(result)
    return result
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.cpp
#include <algorithm>
#include <iostream>
#include <map>
#include <vector>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    map<string, int> m;

    for (int i = 0; i < name.size(); i++) {
        m[name[i]] = yearning[i];
    }

    for (vector<string> v : photo) {
        int sum = 0;
        for (string name : v) {
            sum += m[name];
        }
        answer.push_back(sum);
    }

    return answer;
}